tests/virt_autotest/kubevirt_tests_server.pm

```diff
@@ -84,15 +99,22 @@ sub rke2_server_setup {
     my ($self, $agent_ip) = @_;
 
     record_info('Start RKE2 server setup', '');
-    systemctl('stop apparmor.service');
-    systemctl('stop firewalld.service');
+    unless (is_sle_micro) {
+        disable_and_stop_service('apparmor.service');
+        disable_and_stop_service('firewalld.service');
+    }
     $self->setup_passwordless_ssh_login($agent_ip);
 
+    # rebootmgr has to be turned off as prerequisity for this to work
+    script_run('rebootmgrctl set-strategy off') if (is_sle_micro);
     # Check if the package 'ca-certificates-suse' are installed on the node
     ensure_ca_certificates_suse_installed();
+    transactional::process_reboot(trigger => 1) if (is_sle_micro);
 
-    # Install downstream kubevirt packages
-    zypper_call('in kubernetes1.18-client') if (script_run('rpmquery kubernetes1.18-client'));
+    unless (is_sle_micro) {
+        # Install downstream kubevirt packages
+        zypper_call('in kubernetes1.18-client') if (script_run('rpmquery kubernetes1.18-client'));
+    }
```

Comment:
The above section ,line 102 until here, seems to be the same with agent.pm. Have you considered wrapping it into a function and share by agent and server?

Reply:
It's actually specified for kubevirt test, not very general for most cases, so I would not encapsulate into a function. In addition, frequent function call will increase the system overhead.

Reply:
okay, fine with it if it is your decision